packages/engine-core/src/library/DefaultLibraryLoader.ts

```diff
@@ -16,9 +17,60 @@ import { Library, LibraryLoader } from './types/Library'
 
 const debug = Debug('prisma:client:libraryEngine:loader')
 
-export function load<T>(id: string): T {
-  // this require needs to be resolved at runtime, tell webpack to ignore it
-  return eval('require')(id) as T
+const libraryCacheSymbol = Symbol('PrismaLibraryEngineCache')
+
+type LibraryCache = Record<string, Library | undefined>
+
+type GlobalWithCache = typeof globalThis & {
+  [libraryCacheSymbol]?: LibraryCache
+}
+
+function getLibraryCache(): LibraryCache {
+  const globalWithCache = globalThis as GlobalWithCache
+  if (globalWithCache[libraryCacheSymbol] === undefined) {
+    globalWithCache[libraryCacheSymbol] = {}
+  }
+  return globalWithCache[libraryCacheSymbol]
+}
+
+export function load(libraryPath: string): Library {
+  const cache = getLibraryCache()
+
+  if (cache[libraryPath] !== undefined) {
+    return cache[libraryPath]!
+  }
+
+  // `toNamespacedPath` is required for native addons on Windows, but it's a no-op on other systems.
+  // We call it here unconditionally just like `.node` CommonJS loader in Node.js does.
+  const fullLibraryPath = path.toNamespacedPath(libraryPath)
+  const libraryModule = { exports: {} as Library }
+
+  let flags = 0
+
+  if (process.platform !== 'win32') {
+    // Add RTLD_LAZY on Unix. This is what Node.js does by default
+    // if no flags were passed to dlopen from JavaScript side.
+    //
+    // @ts-expect-error TODO: typings don't define dlopen -- needs to be fixed upstream
+    flags |= os.constants.dlopen.RTLD_LAZY
+  }
+
+  if (process.platform === 'linux') {
```

Comment:
Is the issue also happening on other platforms (bsd, darwin)? I think `RTLD_DEEPBIND` is supported on BSDs.

Reply:
Good point.

On Darwin the issue doesn't happen for a few reasons:
* We don't dynamically link to OpenSSL on macOS: if we use it there at all, then it must be statically linked
* I'm not sure we even use it there though: we probably rely on Apple's `Security.framework` via [`native_tls`](https://docs.rs/native-tls/latest/native_tls) crate
* If we used it, and we dynamically linked to it, the issue still wouldn't happen because macOS resolves symbols differently and uses something called [two-level namespaces](https://developer.apple.com/library/archive/documentation/DeveloperTools/Conceptual/MachOTopics/1-Articles/executing_files.html), which should prevent this issue.

There should be no harm (or, as far as I understand it, even any difference in behaviour) using the flag on Darwin though.

However, on other platforms like BSD the issue may still happen I think. And apparently they do support RTLD_DEEPBIND too. It looks like it's a good idea to always use it indeed.

Reply:
> We don't dynamically link to OpenSSL on macOS: if we use it there at all, then it must be statically linked

Good to know!